### ISSUE TITLE
Dead code is not detected after a never returning function call

### ISSUE THREAD

**Issue by 3n-mb** [Microsoft/TypeScript#13958]
Related to #13957 

**TypeScript Version:**  2.1.4
`"strictNullChecks": true`

**Code**

```ts

function justThrow(): never {
	throw new Error('We throw');
}

function makeString(n: number): string {
	justThrow();
	return `This is a dead code`;
}

```

**Expected behavior:**
Report that there is an unreachable (dead) code after a call to *never returning* function.

**Actual behavior:**
Dead code is not detected.

**Comment by RyanCavanaugh** [comment]
Flow control and function type analysis are done in two passes that don't fully interact with each other. Tons of open issues on this already, see #12825 as a starting point.
